### DISCUSSION TITLE
Date/Time functions and SAMPLE BY

### DISCUSSION THREAD

**Discussion by robocoder** [questdb/questdb#1443] [Ideas]
This [example](https://questdb.io/docs/reference/sql/sample-by):
```
SELECT ts, count() FROM trades
SAMPLE BY 1h
```
appears to lose the SAMPLE BY aggregation when a date/time function is applied to `ts`, e.g.,
```
SELECT to_timezone(ts, 'America/Toronto'), count() FROM trades
SAMPLE BY 1h
```

Is this intentional?

**Comment by bluestreak01** [comment]
timezone alignment is built into the `sample by` clause, for example you could try:

```
SELECT to_timezone(ts, 'America/Toronto'), count() FROM trades
SAMPLE BY 1h
ALIGN TO CALENDAR TIME ZONE 'America/Toronto'
```

**Reply by robocoder** [reply]
In your example:
* the `to_timezone()` function causes the query to no longer aggregate using `SAMPLE BY`.
* the `ALIGN TO CALENDAR TIME ZONE` clause doesn't change how `ts` appears in the results (i.e., still UTC)
* a `WITH OFFSET` clause is required (which seems contrary to the docs)
